### DISCUSSION TITLE
A separate context API is being used for each app page router.

### DISCUSSION THREAD

**Discussion by sanghogu** [vercel/next.js#67679] [Help]
### Summary

app/dashobard page root = DashboardContext

app/order page root = OrderContext

When using it this way, all sub-elements except those that are absolutely necessary use 'use server'.

Is it a bad pattern to use it like this?
What is a safe and good way to manage state in nextjs?

I know that the disadvantage of using the context API with CRA is that re-rendering occurs in all elements under state management.

thanks!

**Comment by icyJoseph** [comment]
I think you are confused here. `use server` is a way for client code to invoke server side code. It is not a way to define components are server components. Components are server components by default, unless, they have `use client`, or are owned by a React sub tree, that uses `use client`.

I am also not sure about the problem you are facing. I think you mean, that you want to share state between pages. You can do that, by placing a component that, internally uses context provider (and thus has `use client`), at the root layout. There is a a key concept to understand, which is the difference between a tree parent, and an owner parent. Otherwise, known as composition: https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props
